internal/provider/cluster_resource.go

```diff
@@ -794,15 +794,22 @@ func loadClusterToTerraformState(
 			RoutingId: types.StringValue(clusterObj.Config.Serverless.RoutingId),
 		}
 
-		// Set either the spend limit or usage limits, depending on the cluster
-		// object requested. Both options are returned by the API.
-		if clusterObj.Config.Serverless.UsageLimits != nil {
+		// The whole plan or just its serverless config can be nil.
+		var planConfig *ServerlessClusterConfig
```

Comment:
Consider doing something like this to avoid the complicated-looking conditional logic:
```
forceSpendLimit := false
if plan != nil && plan.ServerlessConfig != nil {
  forceSpendLimit = !plan.ServerlessConfig.SpendLimit.IsNull()
}
...
if !forceSpendLimit && clusterObj.Config.Serverless.UsageLimits != nil {
} else if clusterObj.Config.Serverless.SpendLimit != nil {
}
```

Reply:
Unfortunately, that doesn't work in the case where the plan exists but doesn't contain either field.  The API response will say the spend limit is zero, but the state needs to be null to match the plan. I'm sure there's still a way to simplify this, but my brain is in vacation mode :)

Reply:
Yes, you're right, I forgot that the API returns `0/nil` rather than `nil/nil` for this case. I'd probably modify my code above to be like this:
```
forceSpendLimit := false
if plan != nil && plan.ServerlessConfig != nil {
  forceSpendLimit = !plan.ServerlessConfig.SpendLimit.IsNull()
}
...
if !forceSpendLimit && clusterObj.Config.Serverless.UsageLimits != nil {
} else {
  spendLimit := clusterObj.Config.Serverless.SpendLimit
  if spendLimit != nil && *spendLimit != 0 {
  }
}
```